lib/finch/http2/request_stream.ex

```diff
@@ -32,6 +23,9 @@ defmodule Finch.HTTP2.RequestStream do
     }
   end
 
+  defp with_byte_size(binary) when is_binary(binary), do: {binary, byte_size(binary)}
+  defp with_byte_size(io_data), do: io_data |> IO.iodata_to_binary() |> with_byte_size()
```

Comment:
if it's possible to leave this as a binary (and thus avoid copying) and use `IO.iodata_length/1` (still O(n) though) it might be a bit more efficient but I didn't benchmark this.

Reply:
Good point! Thanks @wojtekmach, I’ll try it out

Reply:
I assume you meant "if it is possible to leave this as **iodata**". I tried quickly changing this to: `defp with_byte_size(io_data), do: {io_data, IO.iodata_length(io_data)}` but we are currently relying on binary pattern matching to extract binaries that will fit within the current window, therefore we do need to cast the iodata to a binary with the current implementation. I don't believe that there is a straightforward way to slice N bytes from iodata, is there?

Reply:
Yeah I meant to leave it as iodata and yeah, don’t think there is a way to efficiently slice it. :(

Reply:
Maybe something like https://github.com/ninenines/cowlib/blob/master/src/cow_iolists.erl could be used?

Reply:
Nice find! We can try doing some benchmarking to see if this is worth it. I wonder if there was ever any discussion around adding this functionality to the stdlib...